Indexing/TermSharing.cpp

```diff
@@ -70,7 +69,7 @@ TermSharing::~TermSharing()
 void TermSharing::setPoly()
 {
   //combinatory superposiiton can introduce polymorphism into a monomorphic problem
```

Comment:
We no longer have "combinatory superposiiton". We should fix the comment at least.

But is it also true for the new HOL calculus that it can bring poly, even if we did not have it?